test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -1558,6 +1558,12 @@ TEST_P(SslSocketTest, MultiCertPickRSAOnSniMatch) {
 
 // On SNI mismatch, if full scan is disabled, validate that the first cert is used.
 TEST_P(SslSocketTest, MultiCertWithFullScanDisabledOnSniMismatch) {
+  // This test is specific for no full scan case, i.e, below flag is true case.
+  // If the flag is false, the test needs to be skipped.
+  if (!Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.no_full_scan_certs_on_sni_mismatch")) {
```

Comment:
Emm... a question, how it can be false, without explicitly override the runtime flag value, like
```
TestScopedRuntime scoped_runtime;
  scoped_runtime.mergeValues(
      {{"envoy.reloadable_features.no_full_scan_certs_on_sni_mismatch", "false"}});
```

Reply:
We simply changed the runtime flag in source/common/runtime/runtime_features.cc to be false by 
FALSE_RUNTIME_GUARD(envoy_reloadable_features_no_full_scan_certs_on_sni_mismatch), then started a blaze build of ssl_socket_test.cc, and this test failed.

Reply:
emm...if you change `source/common/runtime/runtime_features.cc` directly, that means you have fork version envoy. It doesn't make sense change the upstream for your downstream code. This test also ensures the envoy default behavior. With this change, it loses this guarantee.

Reply:
Agree with @soulxu , if you use FALSE_RUNTIME_GUARD, the default behavior is changed.

Reply:
Thanks for the comments. I was under the impression that our test code need to be passing even if we toggle the runtime flag.  If a test is written for a specific runtime flag state, we should either explicitly specify it, or skip the test otherwise.